src/core/dfn-finder.js

```diff
@@ -0,0 +1,255 @@
+import { pub } from "./pubsubhub";
+import { wrapInner } from "./utils";
+
+const topLevelEntities = new Set([
+  "callback interface",
+  "callback",
+  "dictionary",
+  "enum",
+  "interface mixin",
+  "interface",
+  "typedef",
+]);
+
+// TODO: make these linkable somehow.
+// https://github.com/w3c/respec/issues/999
+// https://github.com/w3c/respec/issues/982
+const unlinkable = new Set(["maplike", "setlike", "stringifier"]);
+
+// This function looks for a <dfn> element whose title is 'name' and
+// that is "for" 'parent', which is the empty string when 'name'
+// refers to a top-level entity. For top-level entities, <dfn>
+// elements that inherit a non-empty [dfn-for] attribute are also
+// counted as matching.
+//
+// When a matching <dfn> is found, it's given <code> formatting,
+// marked as an IDL definition, and returned. If no <dfn> is found,
+// the function returns 'undefined'.
+/**
+ * @param {*} defn
+ * @param {string} parent
+ * @param {string} name
+ * @param {Record<string, HTMLElement[]>} definitionMap
+ * @param {HTMLElement} idlElem
+ */
+export function findDfn(defn, parent, name, definitionMap, idlElem) {
+  switch (defn.type) {
+    case "attribute":
+      return findAttributeDfn(defn, parent, name, definitionMap, idlElem);
+    case "operation":
+      return findOperationDfn(defn, parent, name, definitionMap, idlElem);
+    default:
+      return findNormalDfn(defn, parent, name, definitionMap, idlElem);
+  }
+}
+
+/**
+ * @param {*} defn
+ * @param {string} parent
+ * @param {string} name
+ * @param {Record<string, HTMLElement[]>} definitionMap
+ * @param {HTMLElement} idlElem
+ */
+function findAttributeDfn(defn, parent, name, definitionMap, idlElem) {
+  const parentLow = parent.toLowerCase();
+  const asLocalName = name.toLowerCase();
+  const asQualifiedName = parentLow + "." + asLocalName;
+  let dfn;
+  if (definitionMap[asQualifiedName] || definitionMap[asLocalName]) {
```

Comment:
I wonder if this should be:

```JS
if (definitionMap[asQualifiedName]){ 
  dfn = findNormalDfn(defn, parent, asQualifiedName, definitionMap, idlElem);
} else if(definitionMap[asLocalName]){
  dfn = findNormalDfn(defn, parent, asLocalName, definitionMap, idlElem);
} 
if(!dfn) {
  return findNormalDfn(defn, parent, name, definitionMap, idlElem);
}
```

Or does it not matter?

Reply:
`findNormalDfn()` internally does that in line 146 so maybe it's just okay.

Reply:
Ok, cool. We probably **later** fix it so it's doesn't do things like `delete definitionMap[dottedName];` (old behavior)... that seems not great.

Reply:
Should it be `definitionMap[dottedName] = undefined`?

Reply:
IMO, it shouldn't trash definitions... the defintionMap should be treated as immutable.